src/status_im/contexts/wallet/send/input_amount/view.cljs

```diff
@@ -103,31 +107,28 @@
             input-num-value   (parse-double @input-value)
             suggested-routes  (rf/sub [:wallet/wallet-send-suggested-routes])
             route             (rf/sub [:wallet/wallet-send-route])
-            confirm-disabled? (or
-                               (nil? route)
-                               (empty? @input-value)
-                               (<= input-num-value 0)
-                               (> input-num-value (:amount @current-limit)))
+            confirm-disabled? (or (nil? route)
+                                  (empty? @input-value)
+                                  (<= input-num-value 0)
+                                  (> input-num-value (:amount @current-limit)))
             amount            (str @input-value " " token-symbol)]
         (rn/use-effect
          (fn []
            (let [dismiss-keyboard-fn   #(when (= % "active") (rn/dismiss-keyboard!))
                  app-keyboard-listener (.addEventListener rn/app-state "change" dismiss-keyboard-fn)]
              #(.remove app-keyboard-listener))))
-        (rn/use-effect (fn []
-                         (rf/dispatch [:wallet/clean-suggested-routes])
-                         (when-not (or
-                                    (empty? @input-value)
-                                    (<= input-num-value 0)
-                                    (> input-num-value (:amount @current-limit)))
-                           (debounce/debounce-and-dispatch [:wallet/get-suggested-routes @input-value]
-                                                           100)))
-                       [@input-value])
-        [rn/view
-         {:style style/screen}
+        (rn/use-effect
+         (fn []
+           (rf/dispatch [:wallet/clean-suggested-routes])
+           (when-not (or (empty? @input-value)
+                         (<= input-num-value 0)
+                         (> input-num-value (:amount @current-limit)))
```

Comment:
Shouldn't the current limit amount be included in the dependencies' vector?

Reply:
I Just modified the code style here :eyes: 

But here's my guess based on what I read, but I don't fully know the internals of this component, I think @briansztamfater or @J-Son89 are more involved.

My perspective:

This `current-limit` atom is the thing being initialized with the values from the problematic sub, I tried removing it, but noticed it wasn't too different.

`current-limit` is not an atom "directly" updated by the user, it just stores a two key map with `:amount` and `:currency`, it changes its values based on a `crypto?` predicate:

```clojure
 (reset! current-limit (if crypto?
                        {:amount   limit-crypto
                         :currency token-symbol}
                        {:amount   limit-fiat
                         :currency currency}))
```

Since the values of this map aren't dynamic (they are being gathered from subs inside the "on mount" part of a `from-2` component), I think they could just be regular vars:

```clojure
(let [crypto-limit {:amount   limit-crypto
                    :currency token-symbol}
      fiat-limit   {:amount   limit-fiat
                    :currency currency}])
```

And use them based on `crypto?`.

So,  this `use-effect` isn't directly listening to changes in `current-limit`, `current-limit` works like a place to store values, so maybe it's not needed to add it to the dependency vector.

Reply:
It's a lot of context to ask from a dev to know why the amount is not in the deps vector. But it's your call @ulisesmac :)

Side note: I've fixed so many bugs due to missing dependencies in `useEffect` that, by default, I always prefer to add dynamic values used inside `useEffect` in its deps vector. There are even linters forcing this rule. There are exceptions for sure, but I tend to start with this safer practice.